src/spike/index.test.js

```diff
@@ -0,0 +1,40 @@
+const { test } = require('tap')
+
+const createSignedRequest = require('./index.js')
+const extractors = require('./extractors')
+const transformers = require('./transformers')
+const constructSignature = require('./constructSignature')
+const constructSignatureString = require('./constructSignatureString')
+const outputHandler = require('./outputHandler')
+
+test('the thing', ({ fail }) => {
+  const signatureFields = [
+    '(request-target)',
+    'Content-Type',
+    'Digest'
+  ]
+
+  const opts = {
+    secret: 'topsecret',
+    keyId: 'testkey',
+    algorithm: 'sha512',
```

Comment:
This needs to be `hs2019`. The _actual_ algorithm for signing the request should be either another option, or simply pre-baked into the `constructSignature`. To keep the API from bloating too much that might be a good option.